Cargo.toml

```diff
@@ -54,3 +51,7 @@ timerfd = "1"
 
 [target.'cfg(windows)'.dev-dependencies]
 uds_windows = "1"
+
+[patch.crates-io]
+polling = { git = "https://github.com/smol-rs/polling.git", branch = "notgull/unsafe2" }
+async-io = { path = "." }
```

Comment:
why does this patch `async-io` itself?

Reply:
It pulls in `async-net` as a dev dependency, which in turn pulls `async-io`. If the `async-io` version pulled here is the crates.io version it fails because the API of `polling` had a breaking change.